ImageFeed/Auth/AuthViewController.swift

```diff
@@ -26,18 +30,26 @@ final class AuthViewController: UIViewController, WebViewViewControllerDelegate
     }
 
     func webViewViewController(_ vc: WebViewViewController, didAuthenticateWithCode code: String) -> Void {
+        vc.dismiss(animated: true)
+        DispatchQueue.main.async {
+            UIBlockingProgressHUD.show()
+        }
         let oauth: OAuth2Service = OAuth2Service()
         oauth.fetchAuthToken(code: code, completion: { [weak self] result in
             guard let self = self else { return }
+            DispatchQueue.main.async {
+                UIBlockingProgressHUD.show()
+            }
+            let alertService: AlertService = AlertService(controller: self)
```

Comment:
сервисами обычно называют сущности, которые выполняют операции - вычислить что-то, сходить получить и обработать что-то 

А это все-таки презентер 

Ну и не стоит забывать, что можно писать короче:
```suggestion
            let alertService = AlertService(controller: self)
```